src/nvim/highlight_group.c

```diff
@@ -799,31 +799,41 @@ int lookup_color(const int idx, const bool foreground, TriState *const boldp)
 void set_hl_group(int id, HlAttrs attrs, Dict(highlight) *dict, int link_id)
 {
   int idx = id - 1;  // Index is ID minus one.
-
   bool is_default = attrs.rgb_ae_attr & HL_DEFAULT;
-
-  // Return if "default" was used and the group already has settings
-  if (is_default && hl_has_settings(idx, true)) {
-    return;
-  }
+  bool has_set = hl_has_settings(idx, is_default);
 
   HlGroup *g = &hl_table[idx];
   g->sg_cleared = false;
 
   if (link_id > 0) {
-    g->sg_link = link_id;
-    g->sg_script_ctx = current_sctx;
-    g->sg_script_ctx.sc_lnum += SOURCING_LNUM;
-    g->sg_set |= SG_LINK;
-    if (is_default) {
+    if (is_default && (dict->force || g->sg_deflink == 0)) {
       g->sg_deflink = link_id;
       g->sg_deflink_sctx = current_sctx;
       g->sg_deflink_sctx.sc_lnum += SOURCING_LNUM;
     }
+    // Don't allow a link when there already is some highlighting
+    // for the group, unless force set
+    if (!dict->force && has_set) {
+      if (SOURCING_NAME == NULL && !is_default) {
+        emsg(_(e_group_has_settings_highlight_link_ignored));
+      }
```

Comment:
This part should not be done. It is a _deliberate_ difference of the API that links are not somehow less worth than filling in attributes.

Reply:
okay will remove this part.

Reply:
how about now :thinking: